### PR TITLE
Controller Support

### LINKED ISSUE #375: Controller Support
**Is your feature request related to a problem? Please describe.**
Controller Support is a highly recommended feature request made previously. It’s great feature to add for those who prefer controllers over keyboards/mouse.

**Describe the solution you'd like**
Add controller support

**Additional context**
Support for the following:
- [ ] Xbox Controller

### PR DESCRIPTION
**Pull request by Litorom**
Controller Support:

Current Binds
- A = Attack/Select
- B = Back
- X = Inventory
- Y = Craft
- Menu/Start = Pause
- Joystick/DPad = Movement
- DPad (Menus) = Entry Movement

Tested Controllers:

- Xbox

(Binds may change especially with Quest additions and additional keybinds)
closes #375

### CODE REVIEW

File: src/main/java/minicraft/core/io/ControllerHandler.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	private ControllerIndex controllerIndex;
+
+	public ControllerHandler() {
+		controllerManager.initSDLGamepad();
+		controllerIndex = controllerManager.getControllerIndex(0);
+		controllerManager.update();
+		try {
+			Logger.debug("Controller Detected: " + controllerManager.getControllerIndex(0).getName());
+		} catch (ControllerUnpluggedException e) {
+			Logging.CONTROLLER.debug("No Controllers Detected, moving on.");
```
Comment (BenCheung0422) [inline src/main/java/minicraft/core/io/ControllerHandler.java:18]: I think these two could also Logging.CONTROLLER

**Review by BenCheung0422 — CHANGES_REQUESTED** [review]
Why we don't merge these two InputHandler and ControllerHandler into one InputHandler? Using the mapping instead of creating a new class for it. We need it to be able to maintained easier. We could use a small object to indicate the key mappings of both keyboard and controller, then that's all.

File: build.gradle [hunk 2]
```diff
@@ -19,14 +19,21 @@ mainClassName = 'minicraft.core.Game'
...
     implementation 'org.jetbrains:annotations:23.0.0'
-    implementation 'org.json:json:20211205'
-    implementation 'com.konghq:unirest-java:3.13.7'
+    implementation 'org.json:json:20220320'
+    implementation 'com.konghq:unirest-java:3.13.10'
     implementation 'org.tinylog:tinylog-api:2.4.1'
     implementation 'org.tinylog:tinylog-impl:2.4.1'
+    implementation 'com.badlogicgames.gdx:gdx:1.11.0'
+    implementation 'com.badlogicgames.gdx:gdx-box2d:1.11.0'
+    implementation 'com.badlogicgames.gdx:gdx-controllers:1.9.13'
```
Comment (Makkkkus) [inline build.gradle:34]: Do we need these three when you have Jamepad installed?

**Review by Makkkkus — COMMENTED** [review]
Couldn't find anything really bad. I agree with Ben's review, and encourage you too look into that. Other than that, this is very good!

**Comment by Litorom** [comment]
> Why we don't merge these two InputHandler and ControllerHandler into one InputHandler? Using the mapping instead of creating a new class for it. We need it to be able to maintained easier. We could use a small object to indicate the key mappings of both keyboard and controller, then that's all. Also, the controller keys are not able to be changed in KeyInputDisplay.

For merging InputHandler and ControllerHandler, I attempted that, it was a bit problematic. Also in most games, controller binds are never changeable and are pre-defined

File: src/main/java/minicraft/core/io/ControllerHandler.java (hunk 1 above)
Comment (Litorom) [reply src/main/java/minicraft/core/io/ControllerHandler.java:18]: Fixed, I missed this.

File: build.gradle (hunk 2 above)
Comment (Litorom) [reply build.gradle:34]: Yes, they are required for basic controller functions, with this jamepad and it's other dependency works fine.

**Review by BenCheung0422 — APPROVED** [review]

**Comment by BenCheung0422** [comment]
As a placeholder of controller support before 3.0, it would be good enough.

**Pull request merged**